### ISSUE TITLE
RuleID - execute verification test before merge into master

### ISSUE THREAD

**Issue by philljenkins** [IBMa/equal-access#1421]
Dependent on #1417 being completed where Tom will create a script that makes edits (changes all the ruleIDs)

The list of testing activities below builds from #1397 original testing plan.

See PR #1422  (affects 1,484 files changed), PR heading area also includes the actual list of new & old ruleID changes listed.

See [Checker RuleID Consistency](https://ibm.ent.box.com/file/717584034994?s=kldsplaifciighv1eh3o4fygjw59gk3f) **_spreadsheet_** for a log of comments, why changes, rule fail messages, WCAG no., level, and other comments for background information.

Primarily we'll  validate that the baselines before and after the ruleID changes still work; as executed by the Checker automated actions (PR process)

1. - [x] [accessibility-checker/test/baselines](https://github.com/IBMa/equal-access/tree/master/accessibility-checker/test/baselines) - Erick 
2. - [x] [cypress-accessibility-checker/test/baselines/](https://github.com/IBMa/equal-access/blob/master/cypress-accessibility-checker/test/baselines/) - Erick 
3. - [x] [karma-accessibility-checker/boilerplates/basic/test/baselines/](https://github.com/IBMa/equal-access/blob/master/karma-accessibility-checker/boilerplates/basic/test/baselines/HelloWorld.html.json) - Erick
4. - [x] [accessibility-checker-engine/src/v4/rules](https://github.com/IBMa/equal-access/tree/master/accessibility-checker-engine/src/v4/rules) ruleIDs occurrences in all __.ts files. Spot check for any remaining old ruleIDs in the .ts files - Erick
5. - [x] [accessibility-checker-engine/src/v4/rules](https://github.com/IBMa/equal-access/tree/master/accessibility-checker-engine/src/v4/rules) __.ts rule file names - Erick
6. - [x] [accessibility-checker-engine/help-v4/en-US/](https://github.com/IBMa/equal-access/tree/master/accessibility-checker-engine/help-v4/en-US) __.html help file names (referred to in each __.ts file) - Phill
7. - [x] [accessibility-checker-engine/test/v2/checker/accessibility/rules/](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules)  ___ruleunits folder names - Phill
8. - [x] Spot check - occurrences in test__.html file which there is a `var rule = OpenAjax.a11y.getRule("__ruleID__here__")` - Both
9. - [x] [accessibility-checker-extension/src/ts/devtools/DevToolsPanelApp.tsx](https://github.com/IBMa/equal-access/blob/master/accessibility-checker-extension/src/ts/devtools/DevToolsPanelApp.tsx) list of ~14 ruleIDs in the "Keyboard Mode Rules" in the Extension (see issue #1429) - Phill
10. - [x] [Accessibility Checker Rules](https://unpkg.com/accessibility-checker-engine@next/help/rules.html) list is correctly created  - Erick 
11. - [x] [act-report-v2.txt mapping report](https://unpkg.com/accessibility-checker-engine@3.1.47/act-report-v2.txt) runs and correctly matches "before and after". The ACT Test Results (zipped act-report-v2.txt) in the "**_Action - Artifacts_**" is created automatically with each PR/build. - Phill

**Comment by philljenkins** [comment]
@philljenkins Determine why this oldrleID is still remaining.
https://github.com/IBMa/equal-access/blob/rule-remapping/accessibility-checker-engine/src/v4/rules/IBMA_Color_Contrast_WCAG2AA_PV.ts

**Comment by philljenkins** [comment]
Comparing the ACT Test Results (zipped **act-report-v2.txt**) in the "[Action - Artifacts](https://github.com/IBMa/equal-access/actions/runs/4895114185)"
to the previous deployment's report:
[act-report-v2.txt mapping report](https://unpkg.com/accessibility-checker-engine@3.1.47/act-report-v2.txt)

**Comment by philljenkins** [comment]
Three Help files correctly remain after ruleID changes:
[IBMA_Color_Contrast_WCAG2AA_PV.html](https://github.com/IBMa/equal-access/blob/rule-remapping/accessibility-checker-engine/help-v4/en-US/IBMA_Color_Contrast_WCAG2AA_PV.html)
[Rpt_Aria_ArticleRoleLabel_Implicit.html](https://github.com/IBMa/equal-access/blob/rule-remapping/accessibility-checker-engine/help-v4/en-US/Rpt_Aria_ArticleRoleLabel_Implicit.html)
[Rpt_Aria_GroupRoleLabel_Implicit.html](https://github.com/IBMa/equal-access/blob/rule-remapping/accessibility-checker-engine/help-v4/en-US/Rpt_Aria_GroupRoleLabel_Implicit.html)

**Comment by philljenkins** [comment]
Regarding 7. __ruleunits folder names
all are changed correctly except the following folders remain for archive testing purposes

1. [HAAC_Input_HasRequired_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/HAAC_Input_HasRequired_ruleunit)
2. [Hidden_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/Hidden_ruleunit)
3. [IBMA_Color_Contrast_WCAG2AA_PV_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/IBMA_Color_Contrast_WCAG2AA_PV_ruleunit)
4. [Noscript_visible_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/Noscript_visible_ruleunit)
5. [RPT_Media_ImgColorUsage_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/RPT_Media_ImgColorUsage_ruleunit)
6. [RPT_Style_ExternalStyleSheet_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/RPT_Style_ExternalStyleSheet_ruleunit)
7. [Rpt_Aria_ArticleRoleLabel_Implicit_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/Rpt_Aria_ArticleRoleLabel_Implicit_ruleunit)
8. [Rpt_Aria_GroupRoleLabel_Implicit_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/Rpt_Aria_GroupRoleLabel_Implicit_ruleunit)
9. [Rpt_Aria_MultipleGroupRoles_Implicit_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/Rpt_Aria_MultipleGroupRoles_Implicit_ruleunit)
10. [WCAG20_Label_TargetInvisible_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/WCAG20_Label_TargetInvisible_ruleunit)
11. [WCAG21_Focus_Status_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/WCAG21_Focus_Status_ruleunit)
12. [WCAG_Table_Th_Scope_Role_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/WCAG_Table_Th_Scope_Role_ruleunit)

**Comment by philljenkins** [comment]
Regarding 8. Spot check - occurrences in test__.html files... I checked the following and 
found 73 test case files __.html with occurrences of `OpenAjax.a11y.getRule("` in them
such as these ruleunit test folders

1. [a_text_purpose_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/a_text_purpose_ruleunit)
2. [aria_activedescendant_valid_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/aria_activedescendant_valid_ruleunit)
3. [aria_application_label_unique_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/aria_application_label_unique_ruleunit)
4. [aria_attribute_allowed_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/aria_attribute_allowed_ruleunit)
5. [aria_attribute_conflict_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/aria_attribute_conflict_ruleunit)
6. [aria_attribute_required_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/aria_attribute_required_ruleunit)
7. [aria_attribute_value_valid_ruleunit](https://github.com/IBMa/equal-access/tree/rule-remapping/accessibility-checker-engine/test/v2/checker/accessibility/rules/aria_attribute_value_valid_ruleunit)

**Comment by tombrunet** [comment]
@ErickRenteria To test the rule file:
* In the accessibility-checker-engine directory, run `npm run build:help`
* Open the `rules.html` in the `dist/help` directory.

**Comment by ErickRenteria** [comment]
Ready to push to master
